### PR TITLE
SNOW-946900: Add internal parameter in stored proc registration to allow forcing inline code

### PR DESCRIPTION
**Pull request by sfc-gh-jdu**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-946900

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   This is useful for Pandas API to skip code uploading when the stored procedure is registered by us (not users) and we are pretty sure the function won't be very large.

### CODE REVIEW

File: tests/integ/test_stored_procedure.py [hunk 1]
```diff
@@ -1397,3 +1397,20 @@ def return_success(session_):
             )
             return
         raise
+
+
+def test_force_inline_code(session):
```
Comment (sfc-gh-stan) [inline tests/integ/test_stored_procedure.py:1402]: Can this be tested in Snowpandas instead? This addition is only meaningful to Snowpandas code for now.

Comment (sfc-gh-jdu) [reply tests/integ/test_stored_procedure.py:1402]: we can, but that part code will not be covered in snowpark tests. I don't think it's a good practice to leave uncovered code here.

File: tests/integ/test_stored_procedure.py [hunk 2]
```diff
@@ -1397,3 +1397,20 @@ def return_success(session_):
...
+    def f(session: Session) -> int:
+        return len(large_str)
+
+    with session.query_history() as query_history:
+        _ = session.sproc.register(f, packages=["snowflake-snowpark-python"])
+    assert all("AS $$" not in query.sql_text for query in query_history.queries)
+
+    with session.query_history() as query_history:
+        _ = session.sproc.register(
+            f, packages=["snowflake-snowpark-python"], force_inline_code=True
```
Comment (sfc-gh-aalam) [inline tests/integ/test_stored_procedure.py:1414]: I'm wondering if the way to set this param through public api should be with an internal parameter name like `session.sproc.register(..., _force_inline_code=True)`.

Comment (sfc-gh-jdu) [reply tests/integ/test_stored_procedure.py:1414]: This parameter is added to kwargs, so it won't show up in documents and isn't exposed to users. I follow the same approach as we did for `anonymous` parameter.

File: tests/integ/test_stored_procedure.py (hunk 1 above)
Comment (sfc-gh-jdu) [reply tests/integ/test_stored_procedure.py:1402]: sure, I can do (1), but I'm not sure about (2) whether it's a good idea to add a code link to a private repo right now.

File: tests/integ/test_stored_procedure.py (hunk 2 above)
Comment (sfc-gh-aalam) [reply tests/integ/test_stored_procedure.py:1414]: sounds good. 👍

**Review by sfc-gh-aalam — APPROVED** [review]

**Review by sfc-gh-stan — APPROVED** [review]

**Review by sfc-gh-sfan — APPROVED** [review]

**Pull request merged**